src/qt/optionsmodel.cpp

```diff
@@ -96,61 +97,69 @@ void OptionsModel::Init(bool resetSettings)
     if (!settings.contains("theme"))
         settings.setValue("theme", GUIUtil::getDefaultTheme());
 
-    if (!settings.contains("fontFamily"))
-        settings.setValue("fontFamily", GUIUtil::fontFamilyToString(GUIUtil::getFontFamilyDefault()));
-    if (gArgs.SoftSetArg("-font-family", settings.value("fontFamily").toString().toStdString())) {
-        if (GUIUtil::fontsLoaded()) {
-            GUIUtil::setFontFamily(GUIUtil::fontFamilyFromString(settings.value("fontFamily").toString()));
-        }
-    } else {
+    // Font Family
+    if (!settings.contains("fontFamily")) {
+        settings.setValue("fontFamily", GUIUtil::FontRegistry::DEFAULT_FONT.toUtf8());
+    }
+    if (!gArgs.SoftSetArg("-font-family", settings.value("fontFamily").toString().toStdString())) {
         addOverriddenOption("-font-family");
     }
-
-    if (!settings.contains("fontScale"))
-        settings.setValue("fontScale", GUIUtil::getFontScaleDefault());
-    if (gArgs.SoftSetArg("-font-scale", settings.value("fontScale").toString().toStdString())) {
-        if (GUIUtil::fontsLoaded()) {
-            GUIUtil::setFontScale(settings.value("fontScale").toInt());
+    if (GUIUtil::fontsLoaded()) {
+        if (auto font_name = QString::fromStdString(gArgs.GetArg("-font-family", settings.value("fontFamily").toString().toStdString()));
+            GUIUtil::g_font_registry.RegisterFont(font_name) && GUIUtil::g_font_registry.SetFont(font_name)) {
+            GUIUtil::setApplicationFont();
```

Comment:
We call `setApplicationFont()` in `GuiMain()`. Why do we need an extra call here?